src/Compilers/CSharp/Test/Syntax/Parsing/CollectionLiteralParsingTests.cs

```diff
@@ -0,0 +1,1368 @@
+                        }
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void DictionaryWithTypeExpressions()
+        {
+            UsingTree("_ = [A::B: C::D];");
+
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.GlobalStatement);
+                {
+                    N(SyntaxKind.ExpressionStatement);
+                    {
+                        N(SyntaxKind.SimpleAssignmentExpression);
+                        {
+                            N(SyntaxKind.IdentifierName);
+                            {
+                                N(SyntaxKind.IdentifierToken, "_");
+                            }
+                            N(SyntaxKind.EqualsToken);
+                            N(SyntaxKind.CollectionCreationExpression);
+                            {
+                                N(SyntaxKind.OpenBracketToken);
+                                N(SyntaxKind.DictionaryElement);
+                                {
+                                    N(SyntaxKind.AliasQualifiedName);
+                                    {
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "A");
+                                        }
+                                        N(SyntaxKind.ColonColonToken);
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "B");
+                                        }
+                                    }
+                                    N(SyntaxKind.ColonToken);
+                                    N(SyntaxKind.AliasQualifiedName);
+                                    {
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "C");
+                                        }
+                                        N(SyntaxKind.ColonColonToken);
+                                        N(SyntaxKind.IdentifierName);
+                                        {
+                                            N(SyntaxKind.IdentifierToken, "D");
+                                        }
+                                    }
+                                }
+                                N(SyntaxKind.CloseBracketToken);
+                            }
+                        }
+                        N(SyntaxKind.SemicolonToken);
+                    }
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+    }
```

Comment:
> } [](http://example.com/codeflow?start=3&length=2)

Consider testing the following expressions. (And for testing expressions, consider using `UsingExpression(string)` to simplify the tests.)
`[,]`
`[,A]`
`[A,]`
`[,,]`
`-[A]`
`[A] ?? [B]`
`new T?[1]`
`new T[]?[1]`
`new T[]?[1].Length`
`[..]`
`[..e]`
`[..e]`
`[....]`
`[..e..]`
`[..throw e]`
`[..x.y]`
`[..x = y]`
`[..x => y]`
`[..x ? y : z]`
`[..x ?? y]`